### ISSUE TITLE
Microbatching Support

### ISSUE THREAD

**Issue by shs037** [pytorch/opacus#655]
## 🚀 Feature
Support microbatch size > 1, i.e., clipping multiple (instead of one) gradients.

## Motivation
We want to experiment with microbatch size > 1 for some training tasks.

(I understand that microbatch size  > 1 may not improve memory / computation efficiency. This ask is more about algorithm / utility.)

## Pitch
A `num_microbatches` parameter in `make_private`, similar to [tf privacy](https://github.com/tensorflow/privacy/blob/348895a7a38d7f12db3d5d8e1959cf5ff675b87c/tensorflow_privacy/privacy/optimizers/dp_optimizer.py#L93).

**Comment by shs037** [comment]
Thanks a lot! Is it basically like changing a few lines in the function you linked?

**Comment by HuanyuZhang** [comment]
Yeah, I think a hacky solution (without a very careful interface design) should require minimal changes. ["self.grad_samples"](https://github.com/pytorch/opacus/blob/main/opacus/optimizers/optimizer.py#L404) (per_sample_gradient) is a tensor with shape = batch_size* #parameters. You just need to divide it into several microbatches, and take average for each microbatch. Perhaps you will also need to change "scale_grad" (https://github.com/pytorch/opacus/blob/main/opacus/optimizers/optimizer.py#L441) to make sure of the scale.

**Comment by HuanyuZhang** [comment]
This approach might be problematic if you have multiple mini-batches between the two optimizer steps. But I believe it is a very rare situation.
